### PR TITLE
feat: Adds a new option to disable lobby for guests.

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

File: resources/prosody-plugins/mod_token_verification.lua [hunk 1]
```diff
@@ -33,6 +33,13 @@ log("debug",
 	tostring(host), tostring(token_util.appId), tostring(token_util.appSecret),
 	tostring(token_util.allowEmptyToken));
 
+-- option to disable room modification (sending muc config form) for guest that do not provide token
+local disable_room_moderation_on_empty_token;
+local function load_config()
+    disable_room_moderation_on_empty_token = module:get_option_boolean("disable_room_moderation_on_empty_token");
```
Comment (bgrozev) [inline resources/prosody-plugins/mod_token_verification.lua:39]: Is there a default value? Can we prefix these with the module name so we know what they affect? E.g. token_verification_require_token_for_moderation?

Comment (damencho) [reply resources/prosody-plugins/mod_token_verification.lua:39]: Will rename it. Default is nil which evaluates as false...

File: resources/prosody-plugins/mod_token_verification.lua [hunk 2]
```diff
@@ -70,3 +77,30 @@ module:hook("muc-occupant-pre-join", function(event)
...
+    ["iq-set/bare/http://jabber.org/protocol/muc#owner:query"] = "handle_owner_query_set_to_room" ;
+    -- Host room
+    ["iq-set/host/http://jabber.org/protocol/muc#owner:query"] = "handle_owner_query_set_to_room" ;
+} do
+    module:hook(event_name, function (event)
+        local session, stanza = event.origin, event.stanza;
+
+        -- if disable room mod is false we pass it through, default behaviour
+        -- or the request is coming from admin (focus)
+        if not disable_room_moderation_on_empty_token or is_admin(stanza.attr.from) then
```
Comment (bgrozev) [inline resources/prosody-plugins/mod_token_verification.lua:92]: This is hard to reason about (triple negation: not, disable, "empty_token"). If it was `if not require_token_for_moderation ...` it wouldn't even need a comment

**Comment by damencho** [comment]
How about now?

File: resources/prosody-plugins/mod_token_verification.lua [hunk 3]
```diff
@@ -70,3 +77,30 @@ module:hook("muc-occupant-pre-join", function(event)
...
+for event_name, method in pairs {
+    -- Normal room interactions
+    ["iq-set/bare/http://jabber.org/protocol/muc#owner:query"] = "handle_owner_query_set_to_room" ;
+    -- Host room
+    ["iq-set/host/http://jabber.org/protocol/muc#owner:query"] = "handle_owner_query_set_to_room" ;
+} do
+    module:hook(event_name, function (event)
+        local session, stanza = event.origin, event.stanza;
+
+        -- if disable room mod is false we pass it through, default behaviour
```
Comment (bgrozev) [inline resources/prosody-plugins/mod_token_verification.lua:90]: Comment needs an update (or removal)

File: resources/prosody-plugins/mod_token_verification.lua [hunk 4]
```diff
@@ -33,6 +33,13 @@ log("debug",
 	tostring(host), tostring(token_util.appId), tostring(token_util.appSecret),
 	tostring(token_util.allowEmptyToken));
 
+-- option to disable room modification (sending muc config form) for guest that do not provide token
+local require_token_for_moderation;
+local function load_config()
+    require_token_for_moderation = module:get_option_boolean("token_verification_require_token_for_moderation");
```
Comment (bgrozev) [inline resources/prosody-plugins/mod_token_verification.lua:39]: I guess this defaults to `false`?

File: resources/prosody-plugins/mod_token_verification.lua (hunk 3 above)
Comment (damencho) [reply resources/prosody-plugins/mod_token_verification.lua]: done.

**Comment by damencho** [comment]
Lobby tests are work in progress.

**Pull request merged**